src/main/java/org/fundacionjala/at15/katas/langtonant/agustin/Ant.java

```diff
@@ -0,0 +1,115 @@
+package org.fundacionjala.at15.katas.langtonant.agustin;
+
+public class Ant {
+    // TODO: Make Ant Class with constructor and methods
+
+    // variables
+    int xPos;
+    int yPos;
+    // direction
+    String direction;
+    Grid g;
+    boolean isFinished;
+
+    public Ant(int xPos, int yPos, String direction, Grid g) {
+        // constructor
+        this.xPos = xPos;
+        this.yPos = yPos;
+        this.direction = direction;
+        this.g = g;
+    }
+
+    // Getter coordinates
+
+
+    public int getXPos() {
+        return xPos;
+    }
+
+    public int getYPos() {
+        return yPos;
+    }
+
+    // Setters
+
+    public void setXPos(int xPos) {
+        this.xPos = xPos;
+    }
+
+    public void setYPos(int yPos) {
+        this.yPos = yPos;
+    }
+
+    public void setDirection(String direction) {
+        this.direction = direction;
+    }
+
+    public void run() {
+        //TODO: while(!isFinished) determine direction, change tile color
+
+        while (xPos >= 0 && yPos >= 0 && xPos < g.getWidth() && yPos < g.getHeight() && !isFinished) {
+            System.out.println("x: " + xPos + ", " + "y: " + yPos + ", " + "Facing: " + direction + ", " + "Color: " + g.isWhite(xPos, yPos));
+            if (xPos == 0) {
+                isFinished = true;
+            } else if (xPos == g.getWidth()) {
+                isFinished = true;
+            } else if (yPos == 0) {
+                isFinished = true;
+            } else if (yPos == g.getHeight()) {
+                isFinished = true;
+            }
+
+            // direction
+
+            if (direction.equals("NORTH")) {
```

Comment:
The logic of this if-else blocks seem to have a group of instructions that are repeated, you should consider to create a method where the things that change are set using variables/parameters

Reply:
Yes! I was aware of that. Actually, refactoring this part was one of my TODOs. But it seems that I deleted it for a mistake...
 However, refactoring right away! 👍🏼